lib/logger/lib/logger.ex

```diff
@@ -693,6 +693,68 @@ defmodule Logger do
     :gen_event.sync_notify(Logger, :flush)
   end
 
+  @doc """
+  Set logging level for given module.
+
+  This will take priority over the primary level set, so it can be
+  used to increase or decrease verbosity of some parts of the project.
+
+  ## Example
+
+      defmodule Foo do
+        require Logger
+
+        def log, do: Logger.debug("foo")
+      end
+
+      Logger.configure(level: :error)
+      Logger.set_module_level(Foo, :all)
+
+      Foo.log()
+      # This will print the message even if global level is :error
+
+  """
+  @spec set_module_level(module(), level() | :all | :none) :: :ok | {:error, term()}
+  defdelegate set_module_level(mod, level), to: :logger
+
+  @doc """
+  Get logging level for given module.
+
+  Returned value will be the effective value used. If no value
+  was set for given module then it will not be present in
+  the returned list.
+  """
+  @spec get_module_level(module() | [module()]) :: [{module(), level() | :all | :none}]
+  defdelegate get_module_level(mod), to: :logger
+
+  @doc """
+  Set logging level for all modules in given application.
+
+  This is equivalent of doing:
+
+      Logger.set_module_level(Application.spec(app, :modules), level)
+
+  """
+  @spec set_application_level(atom(), level() | :all | :none) :: :ok | {:error, :not_loaded}
+  defdelegate set_application_level(app, level), to: :logger
+
+  @doc """
+  Reset logging level for given module to primary level.
+  """
+  @spec unset_module_level(module() | [module()]) :: :ok
+  defdelegate unset_module_level(module), to: :logger
```

Comment:
I would include a `delete_module_level/0` or a `delete_module_level(:all)` that delegates to [`:logger.unset_module_level/0`](https://erlang.org/doc/man/logger.html#unset_module_level-0). I think being able to reset everything in just one function call is quite handy 🙂

Reply:
`delete_module_level(:all)` has a conflict with a potential module named `:all`. I suggest either `delete_all_module_levels` or `delete_all_module_level` (since there is a `get_all_env`).

Reply:
Added this as well.